crates/sui-types/src/error.rs

```diff
@@ -936,6 +936,20 @@ impl SuiError {
                     _ => false,
                 }
             }
+
+            // Non retryable errors
```

Comment:
ExecutionError: not sure if this ever gets returned to TD
ByzantineAuthoritySuspicion: I think this no longer applies to TD but it is pretty broad, so ok to keep it here.
Another error to add is `UnsupportedFeatureError`.

Reply:
@mwtian you are right regarding the `ExecutionError` , this seems to be related to the `json-rpc` operations for get balance etc. I can't see indeed how this error could be returned here. I'll remove it from the list.